.github/workflows/main.yml

```diff
@@ -0,0 +1,27 @@
+name: DevOps Course CI
+on: 
+  push:
+    # all branches  
+    branches:
+      - '**'
+    paths:
+      - todo_app/**
+      - Dockerfile
+      - pyproject.toml
```

Comment:
Interesting - you've got down the white-listing rather than the suggested black-listing approach; worth considering the trade-offs here momentarily but I don't have strong opinions:
* This way we only trigger the job if relevant files are updated
  * So we're overall being conservative about when we run our pipeline - should allow for saving the effort/overhead of running pipelines generally
  * But if we added a new file/folder that was important to test, it's possible we'd miss that scenario

Reply:
I understand your consideretations I may apply the backlist in the future, I'll keep it in mind